alibi/utils/discretizer.py

```diff
@@ -1,44 +1,44 @@
 import numpy as np
-from typing import Dict, Callable, List
+from typing import Dict, Callable, Tuple, List
 
 
 class Discretizer(object):
 
-    def __init__(self, data: np.ndarray, categorical_features: List[int], feature_names: List[str],
-                 percentiles: List[int] = [25, 50, 75]) -> None:
+    def __init__(self, data: np.ndarray, numerical_features: List[int], feature_names: List[str],
+                 percentiles: Tuple[int] = (25, 50, 75)) -> None:
```

Comment:
Couple of comments :)

1. `Tuple[int]` indicates a tuple of length 1, perhaps you meant `Tuple[int, ...]`?
2. Why specify `Tuple` or `List` at all, instead of just `Sequence[int]` which allows for either?

Reply:
Hi @MarcoGorelli !

Thanks for the comments, very useful!

- Agreed, I'll make that change.
- kwargs should not contain default mutable values (strange things will happen if that gets changed by the function), the intention there was to change that behaviour. This is a situation where I considered using abstract types for typing was not desirable.

Reply:
Happy to help!

The default value shouldn't be mutable - agreed - but is there any reason to not allow someone to pass in a mutable object when calling the function from somewhere else in the codebase?
```Sequence[int] = (25, 50, 75)```
covers both cases.

Reply:
I think the Sequence protocol makes sense here, we should probably adopt that convention across the library (there are a few other places with mutable list type defaults atm).

Reply:
Good discussion guys, we'll do those changes in a separate PR.